src/parser/jobs/nin/Duality.js

```diff
@@ -0,0 +1,58 @@
+import React, {Fragment} from 'react'
+//import {Icon, Message} from 'semantic-ui-react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+//import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+const DUALITY_GCDS = {
+	[ACTIONS.SPINNING_EDGE.id]: 1,
+	[ACTIONS.GUST_SLASH.id]: 1,
+	[ACTIONS.AEOLIAN_EDGE.id]: 0,
+	[ACTIONS.SHADOW_FANG.id]: 1,
+	[ACTIONS.ARMOR_CRUSH.id]: 1,
+}
+
+export default class Duality extends Module {
+	static handle = 'duality'
+	static dependencies = [
+		'suggestions',
+	]
+
+	_badDualityUses = 0
+	_dualityActive = false
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCast(event) {
+		const abilityId = event.ability.guid
+		
+		if (abilityId === ACTIONS.DUALITY.id) {
```

Comment:
Event order is `applybuff` -> `cast` -> `removebuff`, so you can probably use the `combatants` module and just check `combatants.selected.hasStatus(STATUSES.DUALITY.id)`

Reply:
I'll twiddle that and see if I get the same results. I'll definitely be cleaner than my current approach. And while I'm at it - would that also work for checking for the TA status (with the enemies module) in my TA window module? If yes, I should probably switch that as well - my current approach for that doesn't account properly for adds anyway, now that I think about it.

Reply:
It _should_, yeah. I'm a bit rusty on `combatants` and `enemies`, but have a look around. I think `this.enemies.getEntity(event.targetID).hasStatus(...)` might work? (assuming targetID is a mob)